tools/stats/upload_sccache_stats.py

```diff
@@ -0,0 +1,81 @@
+import fileinput
+import sys
+import os
+from typing import Any
+
+from tools.stats.scribe import (
+    schema_from_sample,
+    rds_write,
+    register_rds_schema,
+)
+
+
+def sprint(*args: Any) -> None:
+    print("[sccache_stats]", *args, file=sys.stderr)
+
+
+def parse_value(value: str) -> Any:
+    try:
+        return int(value)
+    except ValueError:
+        if value.endswith(" s"):
+            return float(value[: -len(" s")])
+
+    return value
+
+
+def get_name(name: str) -> str:
+    return name.replace(" ", "_").replace("-", "_").lower()
+
+
+STAT_NAMES = {
+    "compile_requests",
+    "compile_requests_executed",
+    "cache_hits",
+    "cache_misses",
+    "cache_timeouts",
+    "cache_read_errors",
+    "forced_recaches",
+    "cache_write_errors",
+    "compilation_failures",
+    "cache_errors",
+    "non_cacheable_compilations",
+    "non_cacheable_calls",
+    "non_compilation_calls",
+    "unsupported_compiler_calls",
+    "average_cache_write",
+    "average_cache_read_miss",
+    "average_cache_read_hit",
+    "failed_distributed_compilations",
+}
+
+
+if __name__ == "__main__":
+    if os.getenv("IS_GHA", "0") == "1":
+        data = {}
+        if len(sys.argv) == 2:
+            with open(sys.argv[1]) as f:
+                lines = f.readlines()
+        else:
+            lines = list(fileinput.input())
+        for line in lines:
+            line = line.strip()
+            values = [x.strip() for x in line.split("  ")]
+            values = [x for x in values if x != ""]
+            if len(values) == 2:
+                name = get_name(values[0])
+                if name in STAT_NAMES:
+                    data[name] = parse_value(values[1])
+
+        # The data from sccache is always the same so this should be fine, if it
+        # ever changes we will probably need to break this out so the fields
+        # we want are hardcoded
+        register_rds_schema("sccache_stats", schema_from_sample(data))
+
+        rds_write(
+            "sccache_stats", [data], only_on_master=False
+        )  # TODO: disable this once it's been tested
```

Comment:
will this be removed this PR

Reply:
Yeah before landing, it's just so we can see some [test data](https://metrics.pytorch.org/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Metrics%22,%7B%22datasource%22:%22Metrics%22,%22format%22:%22table%22,%22timeColumn%22:%22updated_at%22,%22metricColumn%22:%22none%22,%22group%22:%5B%5D,%22where%22:%5B%5D,%22select%22:%5B%5B%7B%22type%22:%22column%22,%22params%22:%5B%22average_cache_read_hit%22%5D%7D%5D%5D,%22rawQuery%22:true,%22rawSql%22:%22select%20*%20from%20sccache_stats%20limit%2010;%22,%22table%22:%22sccache_stats%22,%22timeColumnType%22:%22timestamp%22%7D%5D)